presto-parser/src/main/java/com/facebook/presto/sql/tree/Cast.java

```diff
@@ -92,6 +91,34 @@ public boolean isTypeOnly()
         return typeOnly;
     }
 
+    /**
+     * We uniformly change the whole type string to lower case,
+     *  except field names which are enclosed in double quotation marks
+     * */
+    private static String transformCase(String input)
```

Comment:
I think we should also validate that the quotes are matched--suppose someone makes a typo, it will cause the error message or output to be strange and unexpected.  I would also double check that one can't escape double quotes inside double quotes (I don't believe this is possible currently).

Reply:
not sure if the unmatched quotes would be caught by ANTLR parser first or not, my guess would be very possible

Reply:
double-check the SqlBase.g4:
```
ROW '(' identifier type (',' identifier type)* ')'

identifier
    : IDENTIFIER             #unquotedIdentifier
    | QUOTED_IDENTIFIER      #quotedIdentifier
    | nonReserved            #unquotedIdentifier
    | BACKQUOTED_IDENTIFIER  #backQuotedIdentifier
    | DIGIT_IDENTIFIER       #digitIdentifier
```
The parser would throw a syntax error if the quote doesn't end properly.

Reply:
I have run some test cases to make sure that ANTLR parse would indeed catch the unmatched quotes and throw a ParsingException.

Besides, seems we have already handled escape double quotes properly, for SqlBase.g4 defined escape double quotes as '""'. Traditional escape double quotes like '\"' would be consider as two chars: '\' and '"', so the parser would handle it.

Reply:
Makes sense!